app/src/main/java/org/openimis/imispolicies/SQLHandler.java

```diff
@@ -755,7 +755,12 @@ public void insertData(String TableName, String[] Columns, String data, String P
                     object = array.getJSONObject(i);
                     ContentValues cv = new ContentValues();
                     for (String c : Columns) {
-                        cv.put(c, object.getString(c));
+                        if (object.getString(c).equalsIgnoreCase("NULL")) {
+                            cv.putNull(c);
+                        }
```

Comment:
This change alone will result in inconsistency in storing the nulls in the database. It has to be applied in other insert method (line 784), update method (line 797).

The other problem is that there is a lot of functionalities that require the string nulls (i.e. getAvailableProducts line 897) and require to be updated as well.

Reply:
can you please carry on with the proposed updates ?

Reply:
Yes, I will take care of this.

Reply:
This issue is moved to backlog to indicate existing problem, but probably won’t be implemented.

There are quite a lot of inconsistencies in the db regarding “null” or NULL writing. Some features require using “null” values in db, so standardizing this will take a lot of effort, with minimal pay-off.


This issue should be solved in the new version of the app, which further diminishes any potential advantages of this fix in the current version.